# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookieManager.idl
# Commit: 336214ad7891
# Full Hash: 336214ad789156999085eb7607fe6a6cbf7bbfe9
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-10 21:44:02
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 453324c772ae -r 336214ad7891 netwerk/cookie/nsICookieManager.idl
--- a/netwerk/cookie/nsICookieManager.idl	Fri Jan 10 08:24:29 2025 +0000
+++ b/netwerk/cookie/nsICookieManager.idl	Fri Jan 10 08:24:29 2025 +0000
@@ -5,6 +5,7 @@
 
 #include "nsISupports.idl"
 #include "nsICookie.idl"
+#include "nsIThirdPartyCookieBlockingExceptionListService.idl"
 
 %{ C++
 namespace mozilla {
@@ -295,4 +296,19 @@
    * Retrieves all the cookies that were created on or after aSinceWhen, order
    * by creation time */
   Array<nsICookie> getCookiesSince(in int64_t aSinceWhen);
+
+
+  /**
+   * Adds a list of exceptions to the third party cookie blocking exception
+   * list.
+   */
+  void addThirdPartyCookieBlockingExceptions(
+    in Array<nsIThirdPartyCookieExceptionEntry> aExcpetions);
+
+  /**
+   * Removes a list of exceptions from the third party cookie blocking
+   * exception list.
+   */
+  void removeThirdPartyCookieBlockingExceptions(
+    in Array<nsIThirdPartyCookieExceptionEntry> aExceptions);
 };